app/workers/carry_over_unsubmitted_applications_worker.rb

```diff
@@ -14,6 +14,10 @@ def unsubmitted_applications_from_earlier_cycle
       .joins(application_choices: :course)
       .where(submitted_at: nil)
       .where('courses.recruitment_cycle_year' => RecruitmentCycle.previous_year)
+      .where(
+        'application_forms.candidate_id NOT IN (:hidden_candidates)',
+        hidden_candidates: Candidate.where(hide_in_reporting: true).select(:id),
```

Comment:
```suggestion
        hidden_candidates: Candidate.where(hide_in_reporting: true).pluck(:id),
```
Minor nit but I think this avoids creating an array of ActiveRecord objects.

Reply:
Rails has your back here! `Candidate.where(hide_in_reporting: true).select(:id)` returns some kind of `ActiveRecord::ProxyWhatever` which is then used by the main query to generate the eventual SQL.

```rb
[5] pry(main)> ApplicationForm.where(candidate_id: Candidate.where('id > 1'))
  ApplicationForm Load (1.0ms)  SELECT "application_forms".* FROM "application_forms" WHERE "application_forms"."candidate_id" IN (SELECT "candidates"."id" FROM "candidates" WHERE (id > 1))
```

Or with `select` (doesn't seem to be needed):

```rb
[7] pry(main)> ApplicationForm.where(candidate_id: Candidate.where('id > 1').select(:id))
  ApplicationForm Load (1.0ms)  SELECT "application_forms".* FROM "application_forms" WHERE "application_forms"."candidate_id" IN (SELECT "candidates"."id" FROM "candidates" WHERE (id > 1))

```

But with `pluck` it's 2 queries:

```rb
[6] pry(main)> ApplicationForm.where(candidate_id: Candidate.where('id > 1').pluck(:id))
   (0.5ms)  SELECT "candidates"."id" FROM "candidates" WHERE (id > 1)
  ApplicationForm Load (0.8ms)  SELECT "application_forms".* FROM "application_forms" WHERE "application_forms"."candidate_id" IN ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15)  [["candidate_id", 2], ["candidate_id", 3], ["candidate_id", 4], ["candidate_id", 5], ["candidate_id", 6], ["candidate_id", 7], ["candidate_id", 8], ["candidate_id", 9], ["candidate_id", 10], ["candidate_id", 11], ["candidate_id", 12], ["candidate_id", 13], ["candidate_id", 14], ["candidate_id", 15], ["candidate_id", 16]]
```

Reply:
nice, thanks!